### DISCUSSION TITLE
How to define the grid power distribution when using the thermal hydraulics module?

### DISCUSSION THREAD

**Discussion by lipxisi** [idaholab/moose#25186] [Q&A Modules: General] [answered]
Holle

I now have the power distribution data in the form of a geometric matrix like following. 
```
     x -10 -9 ... 8 9 10
y
-10   1    2 . . . 3 2 1 # power data
-9
.
.
.8
9
10
```
Which object can I use to add this type  data to the thermal hydraulic moudule. Do I need to fit it as a function?

Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

There are a number of functions that can load this. There's piecewisemultilinear for example.
A Function will be a widely supported format. 

What are the two coordinates?
component number an axial coordinate?

Guillaume
